packages/scripts.nix

```diff
@@ -784,4 +784,25 @@ lib.makeScope pkgs.newScope (scripts: {
     '';
   };
 
+  # This is a script rather than part of packages/containers.nix because we *want* impurity here,
+  # in order to generate a new image digest and a new tag every time the script runs.
+  push-containerd-reproducer = writeShellApplication {
+    name = "push-containerd-reproducer";
+    runtimeInputs = with pkgs; [
+      crane
+      docker
+      skopeo
+    ];
+    text = ''
+      registry="$1"
+      tmpdir=$(mktemp -d)
+      timestamp=$(date +%s)
+      cat e2e/containerd-digest-pinning/Dockerfile
+      docker build --build-arg TIMESTAMP="$timestamp" -t containerd-reproducer:"$timestamp" \
+        ./e2e/containerd-digest-pinning
+      docker save containerd-reproducer:"$timestamp" -o "$tmpdir/image.tar"
+      skopeo copy docker-archive:"$tmpdir/image.tar" oci:"$tmpdir/oci-image" --insecure-policy
+      crane push "$tmpdir/oci-image" "$registry/contrast/containerd-reproducer:$timestamp"
```

Comment:
You can get the digest directly from the OCI layout:

```
jq -r '.manifests[0].digest' </tmp/img/index.json
```

That should help avoid the regexp in the justfile and the dependency on crane (push with skopeo, too).